api/src/stats.py

```diff
@@ -0,0 +1,253 @@
+
+    # run query
+    query = f'''
+    SELECT *
+    FROM county_stats
+    WHERE election_dt = '{elec_dt.strftime("%y/%m/%d")}';
+    '''
+
+    response = get_county_data(cursor, query, state, elec_dt)
+    response.headers.add('Access-Control-Allow-Origin', '*')
+
+    return response
+
+@stats_bp.route('/county_stats/<county>', methods=['GET'])
+def single_county_stats(county):
+    # get required params
+    state = request.args['state'].upper()
+    elec_dt = datetime.strptime(request.args['election_dt'], '%m-%d-%Y')
+
+    # connect to the database
+    mydb = MySQLdb.connect(host=config['DATABASE']['host'],
+        user=config['DATABASE']['user'],
+        passwd=config['DATABASE']['passwd'],
+        db=config[state]['db'], 
+        local_infile = 1)
+
+    cursor = mydb.cursor(MySQLdb.cursors.DictCursor)
+
+    # run query
+    query = f'''
+    SELECT *
+    FROM county_stats
+    WHERE (election_dt = '{elec_dt.strftime("%y/%m/%d")}'
+    AND county = "{county}");
+    '''
+
+    response = get_county_data(cursor, query, state, elec_dt)
+    response.headers.add('Access-Control-Allow-Origin', '*')
+
+    return response
+
+def get_county_data(cursor, query, state, elec_dt):
+    cursor.execute(query)
+
+    rows = cursor.fetchall()
+
+    county_data = []
+
+    # parse each counties data
+    for row in rows:   
+        # string parsing to convert rej_reason into the right form
+        rej_reason = json.loads(row['rej_reason'])
+        rej_reason = process_json(rej_reason) 
+
+        # get demographic stats
+        demo_stats = get_demographics(state, row) 
+
+        # build county entry and put it in the list
+        county_stats = {
+            "county" : row['county'],
+            "election_dt" : elec_dt.strftime("%m/%d/%Y"),
+            "total_rejected" : row['tot_rejected'],
+            "total_cured" : row['tot_cured'],
+            "total_processed" : row['tot_processed'],
+            "rejected_gender" : demo_stats['gender_rej'],
+            "cured_gender" : demo_stats['gender_cur'],
+            "total_gender" : demo_stats['gender_tot'],
+            "rejected_race" : demo_stats['race_rej'],
+            "cured_race" : demo_stats['race_cur'],
+            "total_race" : demo_stats['race_tot'],
+            "rejected_age_group" : demo_stats['age_rej'],
+            "cured_age_group" : demo_stats['age_cur'],
+            "total_age_group" : demo_stats['age_tot'],
+            "ballot_issue_count" : rej_reason,
+        }
+
+        county_data.append(county_stats)
+
+    # build final output
+    ret_dict = {
+        "state" : state,
+        "election_dt" : elec_dt.strftime("%m/%d/%Y"),
+        "county_data" : county_data
+    }
+
+    return jsonify(ret_dict)
+
+# method to get the demographic info from the mysql row
+def get_demographics(state, row):
+
+    ret_dict = {}
+
+    # GA doesn't support these stats so should be null for Georgia
+    if (state != "GA"):
```

Comment:
I think this is okay for now - but when we get more states' data we will probably have to revisit this part to determine which fields should be null per state

Reply:
Yeah, I agree. This was more so just a temporary solution given that we only have two states right now.